### DISCUSSION TITLE
How Do I Get The Value of a FilledButton or Text control Nested Inside of a ListView?

### DISCUSSION THREAD

**Discussion by AaronCatolico** [flet-dev/flet#938] [Q&A] [answered]
Say that I have a ListView that has either a Text or FilledButton nested inside of it, how do I get the value on_click when one of the items inside of the ListView is clicked?

Here's what I'm working with:

```
from flet import *

def main(page:flet.Page):
    page.window_width=400
    page.window_height=500
    page.bgcolor="#333333"
    page.theme_mode="dark"
    page.window_center()
    page.scroll="always"
    #page.update()

    def show_item(i):
        print(i)


    lv = ListView(expand=True, spacing=5)
    page.add(lv)

    for i in range(1, 101):
        lv.controls.append(FilledButton(text=f"# {i}", on_click=lambda _: show_item(i)))
    lv.update()
    page.update()



flet.app(target=main, assets_dir="assets")
```

**Comment by ndonkoHenri** [comment, marked as answer]
Modify the line in the for loop as follows:
```python
lv.controls.append(FilledButton(text=f"# {i}", data=i, on_click=lambda e: show_item(e.control.data)))
```
All Flet controls have a `data` property, which takes any python data type (int, list, dict etc).
So, you store the `i` in there, and get it back using `e.control.data`.

Let me know if it works as expected after this fix.

**Reply by AaronCatolico** [reply]
Thanks brother! Worked like a charm! BTW, where did you get this information from? I couldn't find it in the flet.dev documents. I'd like to look over it. I've never used 'lambda' before, so I wasn't sure how to get it to work. I knew that I was close though lol. Thanks again.

**Reply by ndonkoHenri** [reply]
Glad it resolved your issue. 
Data prop is in the docs: https://flet.dev/docs/controls/#data

**Reply by bobwatcherx** [reply]
> Senang itu menyelesaikan masalah Anda. Prop data ada di dokumen: [ https://flet.dev/docs/controls/#data ](https://flet.dev/docs/controls/#data)

I have one case that I don't know the answer to.  for example when fetching content inside the Row() widget.  let's say there are two text widgets inside Row() how to get value without using button ? example

```
mysample = Row([
  Text("one"),
 Text("two")

])
ElevatedButton("show one",on_click=showtext)

def showtext(e):
  print(mysample.content[0].value) <----- BUT IS NOT WORK . Is there a solution?

```


and when i click the button .  then do a print .  and what appears is one

and can you provide 2 parameter values ​​in the data property of the button

example
```
ElevatedButton("show one",
data = "one","two"

)
```

**Reply by ndonkoHenri** [reply]
Use this instead:
```python
print(mysample.controls[0].value)
```

`Row` has a `controls` property and not a `content` property.

Let me know if it worked?
